src/mpi/coll/transports/gentran/tsp_gentran.c

```diff
@@ -581,6 +581,15 @@ int MPIR_TSP_sched_start(MPIR_TSP_sched_t s, MPIR_Comm * comm, MPIR_Request ** r
 
     MPIR_FUNC_ENTER;
 
+    if (unlikely(sched->total_vtcs == 0)) {
+        if (!sched->is_persistent) {
+            mpi_errno = MPIR_TSP_sched_free(sched);
+            MPIR_ERR_CHECK(mpi_errno);
+        }
+        *req = MPIR_Request_create_complete(MPIR_REQUEST_KIND__COLL);
```

Comment:
Need an error checking for NULL? Otherwise LGTM

Reply:
Let me take a look at our other usages. At the moment, MPIR_Request_create_complete always returns a constant handle value, but we may still check for NULL in case that ever changes.

Reply:
Sorry, not a constant handle, but a pointer into a statically allocated builtin request array.

Reply:
Added error check for `NULL`, since it is often used with this API.